### PR TITLE
 #46 : Ensuring cross-platform functioning of JUnitRunner.java

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: src/main/java/org/spideruci/tacoco/MavenBuildProbe.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.commons.lang.StringUtils;
```
Comment (VijayKrishna) [inline src/main/java/org/spideruci/tacoco/MavenBuildProbe.java]: Nope... Not needed. Just use a simple StringBuffer or something. If needed create a separate helper that does String Join. If you are still inclined on using a library use Google's commons lib Guava - that is more actively developed and used.

File: src/main/java/org/spideruci/tacoco/MavenBuildProbe.java [hunk 2]
```diff
@@ -31,7 +32,8 @@ public MavenBuildProbe(String absolutTargetPath) {
 	@Override
 	public List<String> getClasses() {
 		makeFilter();
-		scanner.setBasedir(targetDir+"/target/test-classes"); //MAVEN TEST CLASS FOLDER
+        String testClassDirectoryPath = StringUtils.join(new String[] {targetDir, "target", "test-classes"}, File.separator);
```
Comment (VijayKrishna) [inline src/main/java/org/spideruci/tacoco/MavenBuildProbe.java]: Use a StringBuffer, Guava or write a custom utility method that does String Join

File: src/main/java/org/spideruci/tacoco/MavenBuildProbe.java [hunk 3]
```diff
@@ -103,14 +105,14 @@ public BuilderType getBuilderType() {
...
                 MavenCli mavenCli = new MavenCli();
-                mavenCli.doMain(new String[] {"dependency:build-classpath", "-Dmdep.outputFile=tacoco.cp"}, targetDir,
+                mavenCli.doMain(new String[]{"dependency:build-classpath", "-Dmdep.outputFile=tacoco.cp"}, targetDir,
                         System.out, System.out);
 			}
 			classpath = new String(Files.readAllBytes(Paths.get(targetDir,"tacoco.cp")))
-					+":"+ targetDir + "/target/test-classes"
-					+":"+ targetDir + "/target/classes";
+					+ File.pathSeparator + StringUtils.join(new String[]{targetDir, "target", "test-classes"}, File.separator)
+					+ File.pathSeparator + StringUtils.join(new String[]{targetDir, "target", "classes"}, File.separator);
```
Comment (VijayKrishna) [inline src/main/java/org/spideruci/tacoco/MavenBuildProbe.java]: Use a StringBuffer, Guava or write a custom utility method that does String Join

**Comment by tariq1890** [comment]
Sure Vijay. I'll make the changes and post the latest pull request

**Comment by tariq1890** [comment]
I will work on a Utility class to take care of the file path creation and classpath string separation. This way I won't reference File.separator and File.pathSeparator multiple times.

**Comment by tariq1890** [comment]
Change submitted Please check commitId dec68e44196954cdf5d3839de80f4e06b537355f

File: src/main/java/org/spideruci/tacoco/MavenBuildProbe.java [hunk 4]
```diff
@@ -103,14 +105,23 @@ public BuilderType getBuilderType() {
...
-                mavenCli.doMain(new String[] {"dependency:build-classpath", "-Dmdep.outputFile=tacoco.cp"}, targetDir,
+                mavenCli.doMain(new String[]{"dependency:build-classpath", "-Dmdep.outputFile=tacoco.cp"}, targetDir,
                         System.out, System.out);
 			}
-			classpath = new String(Files.readAllBytes(Paths.get(targetDir,"tacoco.cp")))
-					+":"+ targetDir + "/target/test-classes"
-					+":"+ targetDir + "/target/classes";
+
+            final String tacocoDependencies = new String(Files.readAllBytes(Paths.get(targetDir, "tacoco.cp")));
+            final String targetPath = new PathBuilder().path(targetDir).path("classes").buildFilePath();
```
Comment (VijayKrishna) [inline src/main/java/org/spideruci/tacoco/MavenBuildProbe.java:117]: How was this even working in the first place?? there should have been a `.path("target")` in this call chain -- and in the call chain in the next line. It was a small error and i fixed it. But i was wondering if it worked for you (if you tested it before sending the pull request)?

File: src/test/java/org/spideruci/tacoco/util/PathBuilderTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package org.spideruci.tacoco.util;
```
Comment (VijayKrishna) [inline src/test/java/org/spideruci/tacoco/util/PathBuilderTest.java:1]: I love tests! But what do these tests really mean? test, test1 test2 are non-informative names that tell me nothing when the tests actually fail. just something to note for future testing in this or in any other project. :)

File: src/main/java/org/spideruci/tacoco/MavenBuildProbe.java (hunk 4 above)
Comment (tariq1890) [inline src/main/java/org/spideruci/tacoco/MavenBuildProbe.java:117]: I did test it.But I used blinky as the system under test (I thought it was safe enough for a sanity test). I guess I didn't get any error as Blinky didn't have any Junit test cases. I'll use another project from here on. Thanks Vijay!

File: src/test/java/org/spideruci/tacoco/util/PathBuilderTest.java (hunk 5 above)
Comment (tariq1890) [inline src/test/java/org/spideruci/tacoco/util/PathBuilderTest.java:1]: You're right. I'll remedy that in my pull request.

Comment (VijayKrishna) [inline src/test/java/org/spideruci/tacoco/util/PathBuilderTest.java:1]: open a new pull request for that.

**Pull request merged**